pandas/io/excel/_xlsxwriter.py

```diff
@@ -235,6 +237,14 @@ def write_cells(
                 style = self.book.add_format(_XlsxStyler.convert(cell.style, fmt))
                 style_dict[stylekey] = style
 
+            if (
+                isinstance(val, datetime.date) or isinstance(val, datetime.datetime)
+            ) and stylekey.startswith("null"):
+                if cell.col not in datetime_cols:
+                    self.set_column_format(sheet_name, startcol + cell.col, None, style)
+                    datetime_cols.append(cell.col)
+                style = None
```

Comment:
why are you setting this?

Reply:
I am setting the `style` to `None` because the cell formatting takes precedence over the column formatting ( `_write_cell `method in the `Worksheet` class) and if I do not do so then the cell formatting will be set and the column formatting will not get reflected for the cell in the file.